src/visuallyCompleteCalculator.ts

```diff
@@ -37,13 +50,15 @@ class VisuallyCompleteCalculator {
   private inViewportMutationObserver: InViewportMutationObserver;
   private inViewportImageObserver: InViewportImageObserver;
 
+  private subscribers = new Set<MetricSubscriber>();
+  private cancellationSubscribers = new Set<CancellationSubscriber>();
+
   // measurement state
   private lastMutation?: TimestampedMutationRecord;
   private lastImageLoadTimestamp = -1;
   private lastImageLoadTarget?: HTMLElement;
-  private subscribers = new Set<MetricSubscriber>();
-  private navigationCount = 0;
-  private activeMeasurementIndex?: number; // only one measurement should be active at a time
+  private navigationStart = 0;
+  private cancellations = new Map<number, Cancellation>();
```

Comment:
If it's not possible to have two concurrent TTVC measurements happening on the same page (and I'm not sure why it would be), then we probably don't need to actually use a Map for this right? You could just remove any cancellation listeners when a new TTVC event is started and then keep a single reference to `Cancellation`, right?

Reply:
Unfortunately, I can't get away with that here.  :(

I don't know if you recall our discussion about cancellation, but this isn't a safe assumption with the current approach.

https://dropbox.slack.com/archives/C02JSBWMYBG/p1649182024381069?thread_ts=1649112762.954039&cid=C02JSBWMYBG

In short, starting a new measurement (i.e. an AJAX navigation) will "cancel" any in-flight measurements.  However, those in-flight measurements will not resolve until the page becomes idle.  In other words, the cancellation does not "resolve" until the page becomes idle.

```
|---------x  Initial page load (result discarded)
   |------x  first AJAX nav (result discarded)
      |---o  final AJAX nav (metric reported)
```

Reply:
That makes sense, but yeah I agree with you on the complexity. It's still probably worth doing though.